update-css/sw.js

```diff
@@ -0,0 +1,108 @@
+var cacheName = 'update-css-2015.1015.1202';
+
+self.addEventListener('install', function(event) {
+  self.skipWaiting();
+
+  // TODO: Comment explaining what would happen here in a real site
+
+  event.waitUntil(caches.open(cacheName).then(function(cache) {
+    return cache.addAll([
+      'style-1.css',
+      'style-2.css',
+    ]).then(getCurrentCSSFilename).then(function(filename) {
+      // Cache whichever CSS file we currently want to serve as
+      // 'style.css'
+      return cache.match(filename).then(function(response) {
+        if (response) {
+          cache.put('style.css', response);
+        }
+      });
+    });
+  }));
+});
+
+self.addEventListener('activate', function() {
+  if (self.clients && self.clients.claim) {
+    self.clients.claim();
```

Comment:
It could be possible (but very rare) for the UA to kill the service worker before finishing claiming. To ensure not to do this:
```js
event.waitUntil(self.clients.claim());
```
In the other hand, both browsers shipping sw already implements `self.clients` and `self.clients.claim` so you can remove the `if (...)` check.